index.bs

```diff
@@ -5638,6 +5639,107 @@ At this time, one [=credential property=] is defined: the [=resident key credent
+               * Let `salt1` be the value of <code>SHA-256(UTF8Encode("WebAuthn PRF") || 0x00 || UTF8Encode({{AuthenticationExtensionsPRFInputs/input1}}))</code>.
+               * If {{AuthenticationExtensionsPRFInputs/input2}} is present, let `salt2` be the value of <code>SHA-256(UTF8Encode("WebAuthn PRF" || 0x00 || UTF8Encode({{AuthenticationExtensionsPRFInputs/input2}}))</code>.
+           1. Set {{AuthenticationExtensionsClientOutputs/prfEnabled}} to the value of `hmac-secret` in the authenticator extensions output. If not present, set {{AuthenticationExtensionsClientOutputs/prfEnabled}} to [FALSE].
+           1. Set {{AuthenticationExtensionsClientOutputs/prfOutputs}} to the PRF outputs, if any.
+
+: Client extension output ([=registration extension|registration=])
+:: <xmp class="idl">
+    partial dictionary AuthenticationExtensionsClientOutputs {
+        boolean prfEnabled;
+        AuthenticationExtensionsPRFOutputs prfOutputs;
+    };
+    </xmp>
+
+    <div dfn-type="dict-member" dfn-for="AuthenticationExtensionsClientOutputs">
+        :   <dfn>prfEnabled</dfn>
+        ::  [TRUE] if, and only if, the two PRFs are available for use with the created credential.
+
+        :   <dfn>prfOutputs</dfn>
+        ::  The results of evaluating the PRF for the inputs given in {{AuthenticationExtensionsClientInputs/prfEval}}. Outputs may not be available during [=registration=]; see comments in {{AuthenticationExtensionsClientInputs/prfEval}}.
+    </div>
+
+: Client extension input ([=authentication extension|authentication=])
+::  <xmp class="idl">
+    partial dictionary AuthenticationExtensionsClientInputs {
+        record<USVString, AuthenticationExtensionsPRFInputs> prfEvalByCredential;
+    };
+    </xmp>
+
+    <div dfn-type="dict-member" dfn-for="AuthenticationExtensionsClientInputs">
+        :   <dfn>prfEvalByCredential</dfn>
+        ::  A record mapping [=credential IDs=] to PRF inputs to evaluate for that credential.
+    </div>
+
+
+: Client extension processing ([=authentication extension|authentication=])
+::
+       1. If {{AuthenticationExtensionsClientInputs/prfEvalByCredential}} is present and {{PublicKeyCredentialRequestOptions/allowCredentials}} is empty, return a {{DOMException}} whose name is “{{NotSupportedError}}”.
+       1. If either {{AuthenticationExtensionsClientInputs/prfEvalByCredential}} is present or {{AuthenticationExtensionsClientInputs/prfEval}} is present, and {{PublicKeyCredentialRequestOptions/userVerification}} has the value {{UserVerificationRequirement/preferred}}, return a {{DOMException}} whose name is “{{NotSupportedError}}”.
+       1. Find any applicable PRF input(s):
+           1. If {{AuthenticationExtensionsClientInputs/prfEvalByCredential}} is present and contains an entry whose key is the [=base64url encoding=] of the [=credential ID=] that will be returned, use the value of that entry.
```

Comment:
>If `prfEvalByCredential` [...] contains an entry whose key is the [...] credential ID that will be returned [...]

Does the client know this at this time? Or will the client issue `authenticatorGetCredential(up=0)` requests until it finds one (or exhausts the options)?

Reply:
The client will need to batch up=0 requests in order to find the credential that will be asserted first. I think clients already need to support this in practice so I don't anticipate it being a problem.